libraries/lib-datahandler/src/test/java/org/eclipse/sw360/datahandler/resourcelists/ResourcePredicateGeneratorTest.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Bosch Software Innovations GmbH, 2018.
+ * Part of the SW360 Portal Project.
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.sw360.datahandler.resourcelists;
+
+import org.eclipse.sw360.datahandler.thrift.components.Component;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ResourcePredicateGeneratorTest {
```

Comment:
I would have liked another test checking the AND-functionality of the predicates. But as tests are rare in other places of the project, this should not block anything :-)